From c8dd57a9e1443e7ae0d85a1af27a8645b0541977 Mon Sep 17 00:00:00 2001
From: Felix Fietkau <nbd@nbd.name>
Date: Thu, 16 Mar 2023 20:10:29 +0530
Subject: [PATCH 167/500] OpenWrt: 910-kobject_uevent.patch

lib: add uevent_next_seqnum()

Signed-off-by: Felix Fietkau <nbd@nbd.name>

Change-Id: Ie79973ad37bdf26f26e525ef2644d579d1351303
Signed-off-by: Ram Chandra Jangir <quic_rjangir@quicinc.com>
(cherry picked from commit 09b09dc559a4fae7221289a3a9b1a8adfdbdf324)
---
 lib/kobject_uevent.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/lib/kobject_uevent.c b/lib/kobject_uevent.c
index 7c44b7ae4c5c..92712eb8bede 100644
--- a/lib/kobject_uevent.c
+++ b/lib/kobject_uevent.c
@@ -179,6 +179,18 @@ static int kobject_action_args(const char *buf, size_t count,
 	return r;
 }
 
+u64 uevent_next_seqnum(void)
+{
+	u64 seq;
+
+	mutex_lock(&uevent_sock_mutex);
+	seq = ++uevent_seqnum;
+	mutex_unlock(&uevent_sock_mutex);
+
+	return seq;
+}
+EXPORT_SYMBOL_GPL(uevent_next_seqnum);
+
 /**
  * kobject_synth_uevent - send synthetic uevent with arguments
  *
-- 
2.34.1

